test/data/livechat/ILiveChatRepository.test.ts

```diff
@@ -0,0 +1,383 @@
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from InternalDataSource', async () => {
+            const name = 'support';
+
+            when(mockedInternal.getDepartmentByName(name)).thenResolve(departmentFactory.build());
+
+            await livechatRepo.getDepartmentByName(name);
+
+            verify(mockedInternal.getDepartmentByName(name)).once();
+        });
+    });
+
+    describe('#eventCloseRoom()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from CacheDataSource', async () => {
+            const livechat = livechatRoomFactory.build();
+
+            when(mockedCache.deleteRoom(livechat)).thenResolve();
+
+            await livechatRepo.eventCloseRoom(livechat);
+
+            verify(mockedCache.deleteRoom(livechat)).once();
+        });
+    });
+
+    describe('#sendMessage()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from InternalDataSource', async () => {
+            const text = 'text from message';
+            const attachments = mock<Array<IMessageAttachment>>();
+            const livechat: ILivechatRoom = livechatRoomFactory.build();
+
+            when(mockedInternal.sendMessage(text, attachments, livechat)).thenResolve('2hSb3rKy8fn5uwWd');
+
+            await livechatRepo.sendMessage(text, attachments, livechat);
+
+            verify(mockedInternal.sendMessage(text, attachments, livechat)).once();
+        });
+    });
+
+    describe('#getRoomByVisitorToken()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from CacheDataSource', async () => {
+            const token = 'jetaEBBr5jhxU3mS';
+            const livechat: Room = roomFactory.build();
+
+            when(mockedCache.getRoomByVisitorToken(token)).thenResolve(livechat);
+
+            await livechatRepo.getRoomByVisitorToken(token);
+
+            verify(mockedCache.getRoomByVisitorToken(token)).once();
+        });
+
+        it('should throw an Error', async () => {
```

Comment:
Give a more appropriate description (something like "should throw an error when the visitor doesn't exists").